Filter transform media document

bkjohnson opened the pull request:
## Description

### Closes https://github.com/department-of-veterans-affairs/va.gov-team/issues/3317

**Note:** A test currently fails because the `alias` is required to be a string according to the raw schema yet we find an entity where it is null (`media.6c3238d2-99a3-4829-bb54-5c86dd69a285.json `)

Similar to how #11353 is getting the `derivitive`, we may want to have the CMS give us a proper `url` rather than relying on `path[0].alias`

Until we have this, I'm not sure if we should allow `alias` to be null in the schema.  All thoughts welcome.  Here is the troublesome entity (by allowing `alias` to be null, we end up with a filename but no real way to get the file):

```
{
    "uuid": [
        {
            "value": "6c3238d2-99a3-4829-bb54-5c86dd69a285"
        }
    ],
    "langcode": [
        {
            "value": "en"
        }
    ],
    "bundle": [
        {
            "target_id": "document",
            "target_type": "media_type",
            "target_uuid": "584b28c1-b73f-43c8-b1e9-02c86ac63d49"
        }
    ],
    "revision_created": [
        {
            "value": "2019-08-26T21:40:06+00:00",
            "format": "Y-m-d\\TH:i:sP"
        }
    ],
    "revision_user": [
        {
            "target_type": "user",
            "target_uuid": "fd71292f-3dc7-409d-b0e8-500204cb005a"
        }
    ],
    "revision_log_message": [],
    "status": [
        {
            "value": true
        }
    ],
    "name": [
        {
            "value": "VA Form 10-10EZ"
        }
    ],
    "thumbnail": [
        {
            "alt": "",
            "title": null,
            "width": null,
            "height": null,
            "target_type": "file",
            "target_uuid": "bd3b1fc6-cb65-4def-9c2e-21cca6aa48d1"
        }
    ],
    "uid": [
        {
            "target_type": "user",
            "target_uuid": "fd71292f-3dc7-409d-b0e8-500204cb005a"
        }
    ],
    "created": [
        {
            "value": "2019-08-26T21:39:39+00:00",
            "format": "Y-m-d\\TH:i:sP"
        }
    ],
    "changed": [
        {
            "value": "2019-08-26T21:40:06+00:00",
            "format": "Y-m-d\\TH:i:sP"
        }
    ],
    "default_langcode": [
        {
            "value": true
        }
    ],
    "revision_translation_affected": [
        {
            "value": true
        }
    ],
    "moderation_state": [],
    "metatag": {
        "value": {
            "title": "| Veterans Affairs",
            "twitter_cards_type": "summary_large_image",
            "og_site_name": "Veterans Affairs",
            "twitter_cards_title": "| Veterans Affairs",
            "twitter_cards_site": "@DeptVetAffairs",
            "og_title": "| Veterans Affairs"
        }
    },
    "path": [
        {
            "alias": null,
            "pid": null,
            "langcode": "en",
            "pathauto": 1
        }
    ],
    "field_document": [
        {
            "display": true,
            "description": "",
            "target_type": "file",
            "target_uuid": "ffc1d671-71c0-4f90-ab45-d99e7a1f1f3d"
        }
    ],
    "field_media_in_library": [
        {
            "value": true
        }
    ],
    "field_media_submission_guideline": [],
    "field_owner": [
        {
            "target_type": "taxonomy_term",
            "target_uuid": "e5820ec7-83b0-4d03-8ebf-ed50fa8cc211"
        }
    ]
}
```

## Testing done

Yup

## Screenshots


## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

bkjohnson committed:
Use different media-document entity

bkjohnson committed:
Run new content model script

bkjohnson committed:
Adjust the schema & transformation for the path

bkjohnson committed:
Adjust transformer to match pages.json structure

bkjohnson committed:
Change test entity output to match transformer

cvalarida commented on src/site/stages/build/process-cms-exports/schemas/raw/media-document.js:
```diff
@@ -0,0 +1,20 @@
+      items: {
+        type: 'object',
+        properties: {
+          alias: { type: 'string' },
```
:thinking: So right now, we require this to be a string because it's how we find the file (document) this entity is for and without it, we have no other way. Right?

If we merge this in, will `npm run build:content:test` finish successfully? If not, we may need to make it nullable for now, and write a ticket to address it when we have the ability to get the URL of the file instead of relying on this alias.

bkjohnson replied:
Yes to your first question.

No to your second - `build:content:test:` will fail.

So, with this change the tests & content build will pass:

![image](https://user-images.githubusercontent.com/2008881/71853264-aa123d00-308f-11ea-8ddc-6723bad306eb.png)


I suppose I'll commit them and then write up an issue for the url, and maybe open a PR for `nullableString` as well :slightly_smiling_face:

bkjohnson committed:
Allow raw alias + transformed url to be null

This isn't ideal - but it should work until the CMS team sets up
tome-sync to export the real url

bkjohnson committed:
Merge branch 'master' into filter-transform-media-document
